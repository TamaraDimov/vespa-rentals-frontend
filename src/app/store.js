import { configureStore } from '@reduxjs/toolkit';
import userSlice from '../redux/reducers/userSlice';
import motorcycleReducer from '../redux/reducers/motorcycleSlice';
import reservationReducer from '../redux/reducers/reservationSlice';

const store = configureStore({
  reducer: {
    motorcycle: motorcycleReducer,
    reservation: reservationReducer,
    user: userSlice,
  },
});

export default store;
