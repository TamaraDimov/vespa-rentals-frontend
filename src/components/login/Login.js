import React, { useState, useEffect } from 'react';
import { useSelector, useDispatch } from 'react-redux';
import { useNavigate } from 'react-router-dom';
import { BiSolidUser } from 'react-icons/bi';
import { MdEmail } from 'react-icons/md';
import { RiLockPasswordFill } from 'react-icons/ri';
import { toast } from 'react-toastify';
import { loginUser, registerUser } from '../../redux/reducers/userSlice';
import logo from '../../assets/logo.png';
import logo1 from '../../assets/vespa.png';
import './Login-style.css';

const Login = () => {
  const [email, setEmail] = useState('');
  const [username, setUsername] = useState('');
  const [password, setPassword] = useState('');
  const [isMember, setIsMember] = useState(true);

  const dispatch = useDispatch();
  const { isLoading, user } = useSelector((store) => store.user);
  const navigate = useNavigate();

  const handleChange = (e) => {
    const { name, value } = e.target;
    if (name === 'email') {
      setEmail(value);
    }
    if (name === 'username') {
      setUsername(value);
    } else if (name === 'password') {
      setPassword(value);
    }
  };

  const onSubmit = (e) => {
    e.preventDefault();
    if (!email || !password || !username) {
      toast.error('Please fill out all fields');
      return;
    }

    if (isMember) {
      dispatch(loginUser({ username, email, password }));
    } else {
      dispatch(registerUser({ username, email, password }));
    }
  };

  const toggleMember = () => {
    setIsMember(!isMember);
  };

  useEffect(() => {
    if (user) {
      setTimeout(() => {
        navigate('/motorcycles');
      }, 500);
    }
  }, [user, navigate]);

  return (
    <div className="bg-img">
      <form onSubmit={onSubmit} className="form">
        <img src={logo} className="logo" alt="logo" />
        <h3 className="form-title">{isMember ? 'Login' : 'Register'}</h3>
        <div className="form-row">
          <div
            className="wrap-input100 validate-input"
            data-validate="Enter username"
          >
            <input
              type="name"
              value={username}
              name="username"
              onChange={handleChange}
              className="input100"
              placeholder="Username"
            />
            <span className="focus-input100" data-placeholder="&#xf207;">
              <BiSolidUser className="input-icon" />
            </span>
          </div>

          <div
            className="wrap-input100 validate-input"
            data-validate="Enter email"
          >
            <input
              type="email"
              value={email}
              name="email"
              onChange={handleChange}
              className="input100"
              placeholder="Email"
            />
            <span className="focus-input100" data-placeholder="&#xf207;">
              <MdEmail className="input-icon" />
            </span>
          </div>
          <div
            className="wrap-input100 validate-input"
            data-validate="Enter password"
          >
            <input
              type="password"
              value={password}
              name="password"
              onChange={handleChange}
              className="input100"
              placeholder="Enter password"
            />
            <span className="focus-input100" data-placeholder="&#xf207;">
              <RiLockPasswordFill className="input-icon" />
            </span>
          </div>
        </div>
        <button
          type="submit"
          className="login100-form-btn btn-block"
          disabled={isLoading}
        >
          {isLoading ? 'Loading' : 'Submit'}
        </button>
      </form>
      <section className="member">
        <div className="motor-logo">
          <img src={logo1} className="logo1" alt="logo1" />
        </div>
        <div className="register-button">
          <p className="member-text">
            {isMember ? 'Not a member yet?' : 'Already a member?'}
            <br />
            <button type="button" onClick={toggleMember} className="member-btn">
              {isMember ? 'Register' : 'Login'}
            </button>
          </p>
        </div>
      </section>
    </div>
  );
};

export default Login;
