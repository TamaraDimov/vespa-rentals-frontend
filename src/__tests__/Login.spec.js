import { create } from 'react-test-renderer';
import { BrowserRouter as Router } from 'react-router-dom';
import { Provider } from 'react-redux';
import store from '../app/store';
import Login from '../components/login/Login';

describe('Login component', () => {
  test('renders correctly', () => {
    const component = create(
      <Provider store={store}>
        <Router>
          <Login />
        </Router>
      </Provider>,
    );
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
