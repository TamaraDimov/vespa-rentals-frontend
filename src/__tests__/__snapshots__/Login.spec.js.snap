// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login component renders correctly 1`] = `
<div
  className="bg-img"
>
  <form
    className="form"
    onSubmit={[Function]}
  >
    <img
      alt="logo"
      className="logo"
      src="logo.png"
    />
    <h3
      className="form-title"
    >
      Login
    </h3>
    <div
      className="form-row"
    >
      <div
        className="wrap-input100 validate-input"
        data-validate="Enter username"
      >
        <input
          className="input100"
          name="username"
          onChange={[Function]}
          placeholder="Username"
          type="name"
          value=""
        />
        <span
          className="focus-input100"
          data-placeholder=""
        >
          <svg
            className="input-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M7.5 6.5C7.5 8.981 9.519 11 12 11s4.5-2.019 4.5-4.5S14.481 2 12 2 7.5 4.019 7.5 6.5zM20 21h1v-1c0-3.859-3.141-7-7-7h-4c-3.86 0-7 3.141-7 7v1h17z"
            />
          </svg>
        </span>
      </div>
      <div
        className="wrap-input100 validate-input"
        data-validate="Enter email"
      >
        <input
          className="input100"
          name="email"
          onChange={[Function]}
          placeholder="Email"
          type="email"
          value=""
        />
        <span
          className="focus-input100"
          data-placeholder=""
        >
          <svg
            className="input-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M0 0h24v24H0z"
              fill="none"
            />
            <path
              d="M20 4H4c-1.1 0-1.99.9-1.99 2L2 18c0 1.1.9 2 2 2h16c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 4l-8 5-8-5V6l8 5 8-5v2z"
            />
          </svg>
        </span>
      </div>
      <div
        className="wrap-input100 validate-input"
        data-validate="Enter password"
      >
        <input
          className="input100"
          name="password"
          onChange={[Function]}
          placeholder="Enter password"
          type="password"
          value=""
        />
        <span
          className="focus-input100"
          data-placeholder=""
        >
          <svg
            className="input-icon"
            fill="currentColor"
            height="1em"
            stroke="currentColor"
            strokeWidth="0"
            style={
              Object {
                "color": undefined,
              }
            }
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M18 8H20C20.5523 8 21 8.44772 21 9V21C21 21.5523 20.5523 22 20 22H4C3.44772 22 3 21.5523 3 21V9C3 8.44772 3.44772 8 4 8H6V7C6 3.68629 8.68629 1 12 1C15.3137 1 18 3.68629 18 7V8ZM16 8V7C16 4.79086 14.2091 3 12 3C9.79086 3 8 4.79086 8 7V8H16ZM11 14V16H13V14H11ZM7 14V16H9V14H7ZM15 14V16H17V14H15Z"
            />
          </svg>
        </span>
      </div>
    </div>
    <button
      className="login100-form-btn btn-block"
      disabled={false}
      type="submit"
    >
      Submit
    </button>
  </form>
  <section
    className="member"
  >
    <div
      className="motor-logo"
    >
      <img
        alt="logo1"
        className="logo1"
        src="vespa.png"
      />
    </div>
    <div
      className="register-button"
    >
      <p
        className="member-text"
      >
        Not a member yet?
        <br />
        <button
          className="member-btn"
          onClick={[Function]}
          type="button"
        >
          Register
        </button>
      </p>
    </div>
  </section>
</div>
`;
